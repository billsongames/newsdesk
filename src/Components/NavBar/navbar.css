.navbar-container {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  background-color: white;
  height: 80px;
  width: 100%;
}

.navbar-logo {
  margin-right: 15px;
  height: 80px;
}

.navbar-categories {
  display: flex;
  flex-direction: row;
  margin-right: 15px;
  height: 80%
}

.navbar-categories-item {
  margin-left: 10px;
  display: inline-block;
}

.navbar-categories-item a {
  text-decoration: none;
  color: black;
}
  
.navbar-categories-item a:visited {
  text-decoration: none;
  color: black;
}
  
.navbar-categories-item :hover {
font-weight: bold;
text-decoration: underline;
}
  
.navbar-categories-item a:active {
  text-decoration: underline;
}

.navbar-button{
height: 80%;
font-size: 20px;
font-weight: bold;
border:none;
padding: 8px 8px 0px 8px;
background-color:white;
transition-duration: 0.2s;
border-bottom: 8px solid white;
border-left: 1px solid grey;
border-right: 1px solid grey;



}
.navbar-button:hover {
border-bottom: 8px solid black;


}

.navbar-button:active {
background-color: grey;
text-decoration: underline;
}  