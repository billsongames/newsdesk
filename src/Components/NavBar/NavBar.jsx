import React, { useState } from "react";

import "./navbar.css"

const NavBar = ({ onNavBarCategoryChange,onSubmit }) => {
    const [search, setSearch] = useState("");

    const handleSearchInput = (event) => {
        setSearch(event.target.value);
    };

    const handleSubmit = (event) => {
        event.preventDefault();
        onSubmit(search)
    };

    return (
    <div className="navbar-container">
        <img className="navbar-logo" src="/logo2.png" alt="logo" />
        <div className="navbar-categories">
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("general")}>General</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("world")}>World</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("national")}>National</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("business")}>Business</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("technology")}>Technology</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("entertainment")}>Entertainment</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("sports")}>Sports</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("science")}>Science</button>
          <button className="navbar-button" onClick={() => onNavBarCategoryChange("health")}>Health</button>
{/*             <li className="navbar-categories-item">General</li>
            <li className="navbar-categories-item">World</li>
            <li className="navbar-categories-item">National</li>
            <li className="navbar-categories-item">Business</li>
            <li className="navbar-categories-item">Technology</li>
            <li className="navbar-categories-item">Entertainment</li>
            <li className="navbar-categories-item">Sports</li>
            <li className="navbar-categories-item">Science</li> */}



          </div>  

            

        <form className='search-form' onSubmit={handleSubmit}>
            <input 
                className='search-form__input' 
                type='text' 
                placeholder='Search' 
                value={search}
                onChange={handleSearchInput}
        />
    
        <button className='search-form__button' type='submit' data-testid='button'>
            Search
        </button>
        </form>
        

    </div>
    );
};

export default NavBar;