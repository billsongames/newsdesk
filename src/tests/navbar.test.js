import React from "react";
import { render, screen } from "@testing-library/react";
import NavBar from "../Components/Test/Navbar"

describe("NavBar", () => {
    it("renders logo and categories", () => {
    render(<NavBar />);

    expect(screen.getByAltText("logo")).toBeInTheDocument();
    expect(screen.getByText("Tech")).toBeInTheDocument();
    expect(screen.getByText("Business")).toBeInTheDocument();
    expect(screen.getByText("Politics")).toBeInTheDocument();
    expect(screen.getByText("Sport")).toBeInTheDocument();
    expect(screen.getByText("Entertainment")).toBeInTheDocument();
    });
});