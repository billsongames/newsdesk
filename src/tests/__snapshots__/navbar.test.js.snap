// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar-container"
  >
    <img
      alt="logo"
      class="navbar-logo"
      src="/logo2.png"
    />
    <ul
      class="navbar-categories"
    >
      <li
        class="navbar-categories-item"
      >
        General
      </li>
      <li
        class="navbar-categories-item"
      >
        World
      </li>
      <li
        class="navbar-categories-item"
      >
        National
      </li>
      <li
        class="navbar-categories-item"
      >
        Business
      </li>
      <li
        class="navbar-categories-item"
      >
        Technology
      </li>
      <li
        class="navbar-categories-item"
      >
        Entertainment
      </li>
      <li
        class="navbar-categories-item"
      >
        Sports
      </li>
      <li
        class="navbar-categories-item"
      >
        Science
      </li>
      <li
        class="navbar-categories-item"
      >
        Health
      </li>
    </ul>
    <form
      class="search-form"
    >
      <input
        class="search-form__input"
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        class="search-form__button"
        data-testid="button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
